src/lib.rs

```diff
@@ -513,6 +513,49 @@ pub struct BufferSlice<'a> {
     size: Option<BufferSize>,
 }
 
+/// Describes the contents of a [`Buffer`] at creation.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum BufferInitContents<'a> {
+    /// Allocate memory, but do not provide values for the contents.
+    Allocation(BufferAddress),
```

Comment:
let's have this as `Allocation { size: BufferAddress, mapped: bool }`, since `mapped_at_creation` would otherwise make no sense for the `CopyData` variant